Restructure front end for new course/assignment/signup flow.

souravtecken opened the pull request:
1. Added signup option in the navbar dropdown.
2. Seperated courseCards for courseAdmin/(Student/CourseMember)
3. Seperated courseView pages for the same.
4. Added requestCourse option in courses page.
5. courseMember can add assignments within a courseCard.
6. Added collegeMembers  reducer (For retrieving list of teaching members in courseAdminCard)

Course component hierarchy:
- CoursesContainer (Conditionally renders either of the following two views)
	- CoursesAdminView (contains handleApproveCourse, handleRejectCourse methods)
		- CourseCardAdmin (Card for course request)
	- CourseView (contains handleActivateCourse, handleRequestCourse methods)
		- CourseCard (Card for teaching members/student, contains handleCreateAssignment method)

collegeMembersReducer:
- Action: FETCHED_PROFS
	- adds list of professor objects (containing ID and name) to store.
- Action: CLEARED_PROFS
	- Clears professors from store.
- Action: FETCHED_STUDENTS
        - adds list of student objects (containing ID and name) to store.
- Action: CLEARED_PROFS
	- Clears professors from store.

souravtecken committed:
Restructure front end for new course/assignment/signup flow.
1. Added signup option in the navbar dropdown.
2. Seperated courseCards for courseAdmin/(Student/CourseMember)
3. Seperated courseView pages for the same.
4. Added requestCourse option in courses page.
5. courseMember can add assignments within a courseCard.
6. Added professor list reducer (For retrieving list of teaching members in courseAdminCard)

Course component hierarchy:
- CoursesContainer (Conditional renders either of the following two views)
	- CoursesAdminView (contains handleApproveCourse, handleRejectCourse methods)
		- CourseCardAdmin (Card for course request)
	- CourseView (contains handleActivateCourse, handleRequestCourse methods)
		- CourseCard (Card for teaching members/student, contains handleCreateAssignment method)

profReducer:
- Action: FETCHED_PROFS
	- adds list of teacher objects (containing ID and name) to store.
- Action: CLEARED_PROFS
	- Clears teachers from store.

souravtecken committed:
accountManagement.js/CoursesAdminView: Minor edits.
1. Added comment, removed tags from user object in response.
2. Updated route used to fetch teacher details.

adityavinodk commented on client/app/components/Pages/Courses/CourseView.js:
```diff
@@ -0,0 +1,231 @@
+                        <h6>Code<sup>*</sup></h6>
+                        <input type="text" className="form-control" placeholder="Code" name="code" value={this.state.code} onChange={this.handleInputChange} required={true} />
+                    </div>
+                    <div className="form-group text-left">
```
Add a field for the user to enter role. That is taken in by the endpoint

adityavinodk reviewed (changes requested):
The courses page isn't loading, please look into that

souravtecken committed:
Added role selector in course request form

adityavinodk reviewed (commented):
Need minor UI flow changes

souravtecken committed:
Merge branch 'restructure' of https://github.com/aniketnk/alcoding-web-portal into restructure

souravtecken committed:
Updated courseAdminCard.
-> Updated to allow adding of different classes with respective teaching members and sections.

souravtecken committed:
Merge branch 'restructure' of https://github.com/aniketnk/alcoding-web-portal into restructure

souravtecken committed:
Merge branch 'restructure' of https://github.com/aniketnk/alcoding-web-portal into restructure

souravtecken committed:
Updated course admin flow, and other minor changes.
-> Added student/prof selector with role in courseAdminCard
-> Improved signup form appearance
-> Added collegeMember reducer for students and professors, replacing previous prof reducer.
-> Added validation on frontend in courseAdmin flow. (For grad year, addition of classes, selecting of teaching members)
-> Minor UI changes
-> TODO: Replace toasts with alerts

souravtecken committed:
Fixed graduatingYear issue:
-> Added graduatingYear to course schema
-> Moved graduatingYear picker for each class from course as a whole
-> Course Admin card shows selected teaching members outside the selector

RRK1000 reviewed (approved):
LGTM
